crates/matrix-sdk-ui/src/timeline/controller/mod.rs

```diff
@@ -1309,6 +1309,20 @@ impl<P: RoomDataProvider, D: Decryptor> TimelineController<P, D> {
             }
         }
     }
+
+    /// Insert a timeline start item at the beginning of the room, if it's
+    /// missing.
+    pub async fn insert_timeline_start_if_missing(&self) {
+        let mut state = self.state.write().await;
+        let mut txn = state.transaction();
+        for item in txn.items.iter() {
+            if item.is_timeline_start() {
```

Comment:
Can we have a `VirtualTimelineItem::TimelineStart` at another position than the first position? I don' think so.
Consequently, this loop is unecessary (and even dangerous as we will loop over all items if there is no `TimelineStart`):

```rust
if self.items.get(0).is_some_and(|item| item.is_timeline_start()) {
    return;
}

txn.items.push_front(…);
txn.commit();
```

Reply:
I was worried that the date separator or the read marker could go before, but maybe you're right and I could add a test that a date separator can't be first.

Reply:
You're checking the first item in the EventHandler. Be sure to be consistent.